### DISCUSSION TITLE
Custom engine?

### DISCUSSION THREAD

**Discussion by blake-mealey** [11ty/eleventy#1566] [Q&A]
Is it possible to create a custom engine? Also, while I'm here, is using the Node API of eleventy a supported feature?

I'm working on a project to generate documentation from a set of CI pipeline definition files (Azure Pipelines YAML templates), and I'm thinking about essentially wrapping an eleventy project in a custom CLI, for which I'd like to use the Node API. I think a neat way of implementing it would be to create a custom engine which parses the YAML files into markdown. Does that seem like a reasonable approach?

Thanks!

**Comment by Ryuno-Ki** [comment]
What are you referring to by „engine”?

#620 has some hints on how to invoke Eleventy programmatically.
YAML could be read into a data directory (e.g. using GrayMatter)
You then could [create pages from data](https://www.11ty.dev/docs/pages-from-data/).

**Reply by blake-mealey** [reply]
Hey, I was getting the word "engine" from this page: https://www.11ty.dev/docs/languages/ although I could have been using the wrong term. Basically I was looking for something that took the YAML files and turned them into an output as if they were a MD file.

Maybe I just want to "create pages from data" as you suggest though. I will look into that some more.

Thanks!

**Reply by Ryuno-Ki** [reply]
Oh, okay, so it's about template languages.
#117 might be helpful for your research, too.

YAML is a superset of JSON from what I know. So the conversion to JSON could be sufficient.
